opengrok-indexer/src/main/java/org/opengrok/indexer/history/GitRepository.java

```diff
@@ -95,6 +96,28 @@
      */
     private static final Version MINIMUM_VERSION = new Version(2, 1, 2);
 
+    /**
+     * This is a static replacement for 'working' field. Effectively, check if git is working once in a JVM
+     * instead of calling it for every GitRepository instance.
+     */
+    private static final LazilyInstantiate<Boolean> GIT_IS_WORKING = LazilyInstantiate.using(
+            () -> isGitWorking(getCommand(GitRepository.class.getCanonicalName(), CMD_PROPERTY_KEY, CMD_FALLBACK)));
+
+    private static boolean isGitWorking(String repoCommand) {
+        Executor exec = new Executor(new String[]{repoCommand, "--version"});
+        if (exec.exec(false) == 0) {
+            final String outputVersion = exec.getOutputString();
+            final String version = outputVersion.replaceAll(".*? version (\\d+(\\.\\d+)*).*", "$1");
+            try {
+                return Version.from(version).compareTo(MINIMUM_VERSION) >= 0;
+            } catch (NumberFormatException ex) {
+                LOGGER.log(Level.WARNING, String.format("Unable to detect git version from %s", outputVersion), ex);
+            }
+        }
+        return false;
+    }
```

Comment:
Would you mind moving this private static method below the constructor at least?